### ISSUE TITLE
let's discuss navigation (again)

### ISSUE THREAD

**Issue by norayr** [diaspora/diaspora#7472]
I have a five years history on diaspora, and I post often.

So finding the post I have been written even two years ago is a hard task.

For that I spend hours scrolling down, and even Firfox page can crash.

I do not suggest an exact solution, but I suggest to discuss this.
We clearly have problems with navigation.

May be adding page numbers at the bottom of the stream would help. Let's say, I can count that I need to go check for the post I need somwhere between pages numbered 50-60.

Or may be there should be an API for post numbers of user's posts.

The other solution is calendar. Or API to search by date.

**Comment by goobertron** [comment]
There was a brief discussion about some of the issues you raise a couple of years ago here: https://discourse.diasporafoundation.org/t/pagination-instead-of-infinite-scrolling/354

I think there have been some others, but I can't remember where.

I think a combination of post filtering and a 'jump to date' feature could work well.

**Comment by akayani** [comment]
What about something like this. 

Implements the iCalendar specification (RFC-5545) in Ruby. This allows for the generation and parsing of .ics files, which are used by a variety of calendaring applications.

https://rubygems.org/gems/icalendar

How To Open an ICS File

ICS calendar files can be used in email clients like Microsoft Outlook, Windows Live Mail, and IBM Notes (formerly known as IBM Lotus Notes), as well as most popular calendar programs, like Google Calendar for web browsers, Apple Calendar (previously called Apple iCal) for iOS mobile devices and Macs, Yahoo! Calendar, Mozilla Lightning Calendar, and VueMinder.

As an example, say you want to subscribe to a holiday calendar like those found on Calendar Labs. Opening one of those ICS files in a program like Microsoft Outlook will import all the events as a new calendar that you can then have overlayed with other events from the other calendars you're using.

However, while using a local calendar like that is useful for things like holidays that won't change all year long, you might instead want to share a calendar with someone else so that the changes anyone makes is then reflected in the other people's calendars, like when setting up meetings or inviting people to events.

To do that, you could store your calendar online with something like Google Calendar so it's both easy to share with others and also simple to edit wherever you are. See Google's Import Events to Google Calendar guide for uploading an ICS file to Google Calendar, which will let you share and edit the .ICS file with others through a unique URL.

<<<

What about something like that? Create an ICS file where
Subject == Heading in post
Meeting length == from initial post to last comment 
limited to x days == an ongoing limit setting

That should give this feature and integrate the calendar software which could form the basis of even planners.

read hash tags from the timeline. If you could view posts by #tag you could pretty much user build from there. #events #mybirthday #job37  

That would be massively useful and fairly simple yet universal.


https://www.lifewire.com/ics-file-2622749

**Comment by akayani** [comment]
Maybe it would be possible to query the whole of your connected spaces and build a complete map beyond just the user's own posts. The map could be rebuild once a day and cached.

Something where the code starts at query >> values >> display but where at some future time values can be manipulated, cached, sorted, sorted etc.

Start with something as simple as possible and ignore any performance issues just know they are going to need to be considered later. 

As well as using the plugin for other things.

**Comment by SuperTux88** [comment]
@akayani No, we don't need an ical solution for posts or whatever complex solution your second comment is (and no, performance issues can't be ignored anyway). As @goobertron already wrote, a simple 'jump to date' should do the trick.

**Comment by norayr** [comment]
then, when you already at some post, may be "next" or "prev" post links are also useful.

**Comment by akayani** [comment]
I never know what date it is I'm looking for, just the sort stuff I was doing at the time. 
Whatever is useful. I'm certainly not in a position to attempt to program it. 

I know it has become seriously useless on Facebook to review an active user's history. 

Seemed like a good to to present such data but WTF would I know.

**Comment by Flaburgan** [comment]
Jump to date is nice if you have an idea about when you posted it. If you're more about the content, then what we need to solve your problem is a search feature.

**Comment by akayani** [comment]
I don't think my idea is that bad. A calendar of tags used in posts, with some filters.

The PITA with that sort of interface is the query time between page views. What I'm thinking is you elect a time period and build all the data for that period then cache it on the client so you can flip between months without delay. 

I come from a Drupal background where something like that is pretty easy. Say it was an actionable idea, how many hours of programming time would we be taking about?

**Comment by Flaburgan** [comment]
@akayani we have all the post in a SQL database with the associated timestamp. "Jump to a date" would only mean `SELECT * FROM posts WHERE timestamp < theDateAskedByTheUser ORDER BY timestamp`. Why would you want to integrate a new standard and add hundreds of lines?!

What we really need to do is to design a user interface and be careful about performances.

**Comment by goobertron** [comment]
> I never know what date it is I'm looking for, just the sort stuff I was doing at the time.

I wasn't thinking that with a 'jump to date' you'd know the specific date a post was made, but if for example you think 'I'm sure I posted that thing some time last summer', you could jump to September 2016 and scroll back from there, which would cut out 9 months' worth of scrolling in one go.

**Comment by norayr** [comment]
i am afraid that 'jump to date' alone won't solve the problem.
how would i jump? shell i see if i have made a post that day, or i just jump, and may land nowhere, and then, need to try to brute force different dates, to see what happens?
here it would be necessary, even if you have landed somewhere, have a possibility to 'fine tune', i. e. to go to next and previous post.

**Comment by Flaburgan** [comment]
>how would i jump? shell i see if i have made a post that day, or i just jump, and may land nowhere, and then, need to try to brute force different dates, to see what happens?
here it would be necessary, even if you have landed somewhere, have a possibility to 'fine tune', i. e. to go to next and previous post.

I didn't think about going straight to the single post view, simply to scroll inside the profile stream to the post at the given date (or the first before the given date). So the page will still be https://diaspora-fr.org/u/fla for example.

**Comment by Flaburgan** [comment]
Btw the way Facebook is dealing with that is by displaying date (and month) next to the stream, so the user can simply click on it, instead of having to enter it.

**Comment by goobertron** [comment]
As Fla says, the idea is not to jump straight to a single post of a certain date, but to jump back in the stream to roughly where the desired post might be, rather than having to scroll all the way back from today, loading every single post since the one you want to find in the process.

**Comment by akayani** [comment]
"Btw the way Facebook is dealing with that is by displaying date (and month) next to the stream, so the user can simply click on it, instead of having to enter it."

And it is a horrible way to not find what you are looking for. We have # to work with.
